test: fix intermittent failure in p2p_getaddr_caching.py

mzumsande opened the pull request:
Fixes #28133

In the consistency check, it's not enough to check that our address/port is unique, only the combination of source and target must be unique. Otherwise, the OS may reuse ports for connections to different `-addrbind`, which was happening in the failed runs.

While at it, the second commit cleans up duplicate `getaddr` messages in `p2p_getaddr_caching.py` that do nothing but generate `Ignoring repeated "getaddr"` log messages (and cleans up some whitespace the python linter complains about).

mzumsande committed:
test: fix intermittent failure in p2p_getaddr_caching

Only the combined addr:port of source and destination
must be unique. If the destination is different, the same addr:port
for the source may be used by the OS.

mzumsande committed:
test: drop duplicate getaddrs from p2p_getaddr_caching

python p2p instances will automatically send a getaddr msg after
connecting, the explicit message was a duplicate that was being ignored.

vasild commented on test/functional/p2p_getaddr_caching.py:
```diff
@@ -70,11 +71,8 @@ def run_test(self):
         cur_mock_time = int(time.time())
         for i in range(N):
             addr_receiver_local = self.nodes[0].add_p2p_connection(AddrReceiver())
-            addr_receiver_local.send_and_ping(msg_getaddr())
```
There is a similar pattern below on lines 108,110,112. Should be possible to remove those too? If yes, then also `from test_framework.messages import msg_getaddr`.

To other reviewers: the `GETADDR` message is sent at the end of `P2PInterface::on_version()`.

mzumsande replied:
yes, I missed those, now I removed them all!

vasild commented on test/functional/test_framework/test_node.py:
```diff
@@ -645,10 +645,11 @@ def add_p2p_connection(self, p2p_conn, *, wait_for_verack=True, **kwargs):
             sockname = p2p_conn._transport.get_extra_info("socket").getsockname()
             our_addr_and_port = f"{sockname[0]}:{sockname[1]}"
-            info = [peer for peer in self.getpeerinfo() if peer["addr"] == our_addr_and_port]
+            dst_addr_and_port =f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```
nit, missing white space after `=`:
```suggestion
            dst_addr_and_port = f"{p2p_conn.dstaddr}:{p2p_conn.dstport}"
```

mzumsande replied:
fixed

vasild reviewed (approved):
ACK e4d37875e6a440748623250d77ec3be523a1d34d

Today I learned something new: it is possible to have duplicate (srcaddr, srcport) for TCP connections if the destination is different. Thank you!

mzumsande commented:
[e4d3787 ](https://github.com/bitcoin/bitcoin/commit/e4d37875e6a440748623250d77ec3be523a1d34d)to 8a20f76: Addressed feedback, thanks!

> Today I learned something new: it is possible to have duplicate (srcaddr, srcport) for TCP connections if the destination is different. Thank you!

I was also surprised by this!

vasild reviewed (approved):
ACK 8a20f765cce2fc0fadf1a2b66b843b67f2d2ae12
